lib/rucio/core/replica.py

```diff
@@ -1728,7 +1727,6 @@ def __cleanup_after_replica_deletion(rse_id, files, session=None):
                 did_condition.append(
                     and_(models.DataIdentifier.scope == parent_scope,
                          models.DataIdentifier.name == parent_name,
-                         models.DataIdentifier.is_open == False,  # NOQA
```

Comment:
As discussed in the dev meeting, this should be made conditional so communities can decide if they want to get rid of empty open datasets. @cserf can you adapt this? (Or maybe submit it in two PRs?)